sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -256,10 +256,23 @@ public static bool IsStringEmpty(this ExpressionSyntax expression, SemanticModel
         }
 
         public static bool IsMethodInvocation(this InvocationExpressionSyntax expression, KnownType type, string methodName, SemanticModel semanticModel) =>
+            expression.NameIs(methodName) &&
             semanticModel.GetSymbolInfo(expression).Symbol is IMethodSymbol methodSymbol &&
             methodSymbol.IsInType(type) &&
             methodName.Equals(methodSymbol.Name, StringComparison.InvariantCulture);
 
+        public static bool IsMethodInvocation(this InvocationExpressionSyntax expression, ImmutableArray<KnownType> types, string methodName, SemanticModel semanticModel) =>

```

Comment:
A tiny maintenance improvement could be to have only one `IsMethodInvocation` implementation with a variable number of `types` (if there no performance drawback)

Reply:
indeed the recommendation (e.g. [article](https://www.dotnetperls.com/params)) is to have overloads for small number of parameters
- this is what [String.Concat](https://referencesource.microsoft.com/#mscorlib/system/string.cs,3045) does